grid-compose/pkg/types/types.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Specs struct {
+	Version  string             `yaml:"version"`
+	Networks map[string]Network `yaml:"networks"`
+	Services map[string]Service `yaml:"services"`
+	Storage  map[string]Storage `yaml:"storage"`
+}
+
+type Network struct {
+	Type string `yaml:"type"`
+}
+
+type Storage struct {
+	Type string `yaml:"type"`
+	Size string `yaml:"size"`
+}
+
+type Service struct {
+	Flist       string       `yaml:"flist"`
+	Entrypoint  string       `yaml:"entrypoint,omitempty"`
+	Environment KVMap        `yaml:"environment"`
+	Resources   Resources    `yaml:"resources"`
+	NodeID      uint         `yaml:"node_id"`
+	Volumes     []string     `yaml:"volumes"`
+	Networks    []string     `yaml:"networks"`
+	HealthCheck *HealthCheck `yaml:"healthcheck,omitempty"`
+	DependsOn   []string     `yaml:"depends_on,omitempty"`
+}
+
+type HealthCheck struct {
+	Test     []string `yaml:"test"`
+	Interval string   `yaml:"interval"`
+	Timeout  string   `yaml:"timeout"`
+	Retries  int      `yaml:"retries"`
```

Comment:
think of the type used, is that retries going to be a negative at anypoint? use uint or uint8 instead